src/ILLink.RoslynAnalyzer/DataFlow/DynamicallyAccessedMembersBinder.cs

```diff
@@ -0,0 +1,438 @@
+					foreach (var m in members)
+						yield return m;
+				}
+			}
+
+			if (memberTypes.HasFlag (DynamicallyAccessedMemberTypes.PublicNestedTypes)) {
+				foreach (var nested in typeDefinition.GetNestedTypesOnType (filter: null, bindingFlags: BindingFlags.Public)) {
+					yield return nested;
+					var members = new List<ISymbol> ();
+					nested.GetAllOnType (context, declaredOnly: false, members);
+					foreach (var m in members)
+						yield return m;
+				}
+			}
+
+			if (memberTypes.HasFlag (DynamicallyAccessedMemberTypes.NonPublicProperties)) {
+				foreach (var p in typeDefinition.GetPropertiesOnTypeHierarchy (context, filter: null, bindingFlags: BindingFlags.NonPublic | declaredOnlyFlags))
+					yield return p;
+			}
+
+			if (memberTypes.HasFlag (DynamicallyAccessedMemberTypes.PublicProperties)) {
+				foreach (var p in typeDefinition.GetPropertiesOnTypeHierarchy (context, filter: null, bindingFlags: BindingFlags.Public | declaredOnlyFlags))
+					yield return p;
+			}
+
+			if (memberTypes.HasFlag (DynamicallyAccessedMemberTypes.NonPublicEvents)) {
+				foreach (var e in typeDefinition.GetEventsOnTypeHierarchy (context, filter: null, bindingFlags: BindingFlags.NonPublic | declaredOnlyFlags))
+					yield return e;
+			}
+
+			if (memberTypes.HasFlag (DynamicallyAccessedMemberTypes.PublicEvents)) {
+				foreach (var e in typeDefinition.GetEventsOnTypeHierarchy (context, filter: null, bindingFlags: BindingFlags.Public | declaredOnlyFlags))
+					yield return e;
+			}
+
+			if (memberTypes.HasFlag (DynamicallyAccessedMemberTypesOverlay.Interfaces)) {
+				foreach (var i in typeDefinition.GetAllInterfaceImplementations (context, declaredOnly))
+					yield return i;
+			}
+		}
+		public static IEnumerable<IMethodSymbol> GetConstructorsOnType (this ITypeSymbol type, Func<IMethodSymbol, bool>? filter, BindingFlags? bindingFlags = null)
+		{
+			foreach (var member in type.GetMembers () ) {
+				if (!(member is IMethodSymbol method))
+					continue;
+				if (method.MethodKind != MethodKind.Constructor)
+					continue;
+
+				if (filter != null && !filter (method))
+					continue;
+
+				if ((bindingFlags & (BindingFlags.Instance | BindingFlags.Static)) == BindingFlags.Static && !method.IsStatic)
+					continue;
+
+				if ((bindingFlags & (BindingFlags.Instance | BindingFlags.Static)) == BindingFlags.Instance && method.IsStatic)
+					continue;
+
+				if ((bindingFlags & (BindingFlags.Public | BindingFlags.NonPublic)) == BindingFlags.Public && method.DeclaredAccessibility != Accessibility.Public)
+					continue;
+
+				if ((bindingFlags & (BindingFlags.Public | BindingFlags.NonPublic)) == BindingFlags.NonPublic && method.DeclaredAccessibility == Accessibility.Public)
+					continue;
+
+				yield return method;
+			}
+		}
+
+		public static IEnumerable<IMethodSymbol> GetMethodsOnTypeHierarchy (this ITypeSymbol thisType, OperationAnalysisContext context, Func<IMethodSymbol, bool>? filter, BindingFlags? bindingFlags = null)
+		{
```

Comment:
Code is complaining that we don't use the context for these methods, linker used it to calculate the base type the analyzer simply doesn't need it.

Reply:
Thanks